#VRML_SIM R2019a utf8
WorldInfo {
}
Viewpoint {
  orientation -0.8657724185513476 0.4558466626641 0.20649924797381866 0.9337449169958161
  position 1.004645200777125 2.155235704344838 1.7022284577830202
}
TexturedBackground {
}
TexturedBackgroundLight {
}
Floor {
}
SoccerBall {
}
DEF OMNI_WHEELS Robot {
  translation 0.4565519752033118 0.058867677377151484 0.2235756790288992
  rotation 5.03172545240278e-06 0.9999995160228629 -0.0009838336859811238 -2.4537032497350193
  children [
    DEF SOLID1 Solid {
      children [
        DEF WHEEL1 HingeJoint {
          jointParameters HingeJointParameters {
            position -78.13970611593231
            axis 0 0 1
            anchor 0 0 0.1
          }
          device [
            RotationalMotor {
              name "motor_1"
              maxVelocity 30.3688
              maxTorque 0.847386
            }
            PositionSensor {
              name "position_1"
              resolution 0.00384841
            }
          ]
          endPoint Solid {
            translation 0 0 0.1
            rotation 0 0 -1 2.741482429777275
            children [
              DEF sr1 HingeJoint {
                jointParameters HingeJointParameters {
                  position -1190.1906711020663
                  anchor 0 0.05 0.02
                }
                endPoint Solid {
                  translation 0 0.05 0.02
                  rotation 1 0 0 3.614537262055141
                  children [
                    DEF r1 Transform {
                      rotation 0 0 1 1.5708
                      children [
                        DEF SMALL_WHEEL_SHAPE Shape {
                          appearance PBRAppearance {
                            baseColor 0.8 0 0
                            roughness 1
                            metalness 0
                          }
                          geometry Cylinder {
                            height 0.02
                            radius 0.01
                          }
                        }
                      ]
                    }
                  ]
                  name "sr1"
                  boundingObject USE r1
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr2 HingeJoint {
                jointParameters HingeJointParameters {
                  position -3916.379234636796
                  axis 0.707107 0.707107 0
                  anchor -0.03535 0.03535 0.02
                }
                endPoint Solid {
                  translation -0.03535 0.03535 0.02
                  rotation 0.7071067811865476 0.7071067811865476 0 4.328397043266062
                  children [
                    DEF r2 Transform {
                      rotation 0 0 1 2.27
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr2"
                  boundingObject USE r2
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr3 HingeJoint {
                jointParameters HingeJointParameters {
                  position 3229.4795251589853
                  anchor 0 -0.05 0.02
                }
                endPoint Solid {
                  translation 0 -0.05 0.02
                  rotation -1 0 0 0.07772273132212651
                  children [
                    DEF r3 Transform {
                      rotation 0 0 1 1.5708
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr3"
                  boundingObject USE r3
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr4 HingeJoint {
                jointParameters HingeJointParameters {
                  position 3283.571372694898
                  axis 0.707107 0.707107 0
                  anchor 0.03535 -0.03535 0.02
                }
                endPoint Solid {
                  translation 0.03535 -0.03535 0.02
                  rotation 0.7071067811865475 0.7071067811865475 0 3.7486423471538046
                  children [
                    DEF r4 Transform {
                      rotation 0 0 1 2.27
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr4"
                  boundingObject USE r4
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr5 HingeJoint {
                jointParameters HingeJointParameters {
                  position -3177.2153711021306
                  axis 0 1 0
                  anchor -0.05 0 0.02
                }
                endPoint Solid {
                  translation -0.05 0 0.02
                  rotation 0 1 0 2.076394330740182
                  children [
                    DEF r5 Transform {
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr5"
                  boundingObject USE r5
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr6 HingeJoint {
                jointParameters HingeJointParameters {
                  position 3243.675264335583
                  axis 0 1 0
                  anchor 0.05 0 0.02
                }
                endPoint Solid {
                  translation 0.05 0 0.02
                  rotation 0 1 0 1.5516458309164294
                  children [
                    DEF r6 Transform {
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr6"
                  boundingObject USE r6
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr7 HingeJoint {
                jointParameters HingeJointParameters {
                  position -7000.2277485676805
                  axis -0.707107 0.707107 0
                  anchor -0.03535 -0.03535 0.02
                }
                endPoint Solid {
                  translation -0.03535 -0.03535 0.02
                  rotation 0.7071067811865475 -0.7071067811865475 0 0.7593163696211291
                  children [
                    DEF r7 Transform {
                      rotation 0 0 1 0.76
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr7"
                  boundingObject USE r7
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr8 HingeJoint {
                jointParameters HingeJointParameters {
                  position 1728.3072259821408
                  axis -0.707107 0.707107 0
                  anchor 0.03535 0.03535 0.02
                }
                endPoint Solid {
                  translation 0.03535 0.03535 0.02
                  rotation 0.7071067811865475 -0.7071067811865475 0 5.851918799425032
                  children [
                    DEF r8 Transform {
                      rotation 0 0 1 0.76
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr8"
                  boundingObject USE r8
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF WHEEL_TRANS Transform {
                translation 0 0 0.02
                rotation 1 0 0 1.5708
                children [
                  DEF WHEEL_SHAPE Shape {
                    appearance PBRAppearance {
                      baseColor 0.1803921568627451 0.20392156862745098 0.21176470588235294
                      roughness 1
                      metalness 0
                    }
                    geometry Cylinder {
                      height 0.02
                      radius 0.05
                      subdivision 20
                    }
                  }
                ]
              }
            ]
            boundingObject Transform {
              translation 0 0 0.02
              rotation 1 0 0 1.5708
              children [
                DEF wheel1 Shape {
                  appearance PBRAppearance {
                    baseColor 0.8 0.8 0.8
                    transparency 0.12999999523162842
                    roughness 1
                    metalness 0
                    emissiveColor 0 1 0
                  }
                  geometry Cylinder {
                    height 0.02
                    radius 0.05
                  }
                }
              ]
            }
            physics Physics {
            }
          }
        }
      ]
      name "wheel 1"
      boundingObject DEF DUMMY Sphere {
        radius 0.01
      }
      physics Physics {
      }
    }
    DEF SOLID2 Solid {
      rotation 0 1 0 2.09
      children [
        DEF WHEEL2 HingeJoint {
          jointParameters HingeJointParameters {
            position 80.55501840689395
            axis 0 0 1
            anchor 0 0 0.1
          }
          device [
            RotationalMotor {
              name "motor_2"
              maxVelocity 30.3688
              maxTorque 0.847386
            }
            PositionSensor {
              name "position_2"
              resolution 0.00384841
            }
          ]
          endPoint Solid {
            translation 0 0 0.1
            rotation 0 0 1 5.159200530738687
            children [
              DEF sr1 HingeJoint {
                jointParameters HingeJointParameters {
                  position 113.14408860086532
                  anchor 0 0.05 0.02
                }
                endPoint Solid {
                  translation 0 0.05 0.02
                  rotation 1 0 0 0.04675307163276007
                  children [
                    USE r1
                  ]
                  name "sr1"
                  boundingObject USE r1
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr2 HingeJoint {
                jointParameters HingeJointParameters {
                  position -739.4494166172074
                  axis 0.707107 0.707107 0
                  anchor -0.03535 0.03535 0.02
                }
                endPoint Solid {
                  translation -0.03535 0.03535 0.02
                  rotation 0.7071067811865476 0.7071067811865476 0 1.9664496299838423
                  children [
                    USE r2
                  ]
                  name "sr2"
                  boundingObject USE r2
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr3 HingeJoint {
                jointParameters HingeJointParameters {
                  position -1323.8030371110597
                  anchor 0 -0.05 0.02
                }
                endPoint Solid {
                  translation 0 -0.05 0.02
                  rotation -1 0 0 4.334122603346506
                  children [
                    USE r3
                  ]
                  name "sr3"
                  boundingObject USE r3
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr4 HingeJoint {
                jointParameters HingeJointParameters {
                  position -1496.175895838694
                  axis 0.707107 0.707107 0
                  anchor 0.03535 -0.03535 0.02
                }
                endPoint Solid {
                  translation 0.03535 -0.03535 0.02
                  rotation -0.7071067811865476 -0.7071067811865476 0 0.7777927299524506
                  children [
                    USE r4
                  ]
                  name "sr4"
                  boundingObject USE r4
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr5 HingeJoint {
                jointParameters HingeJointParameters {
                  position -512.3979225193433
                  axis 0 1 0
                  anchor -0.05 0 0.02
                }
                endPoint Solid {
                  translation -0.05 0 0.02
                  rotation 0 0.9999999999999999 0 2.8232726693827943
                  children [
                    USE r5
                  ]
                  name "sr5"
                  boundingObject USE r5
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr6 HingeJoint {
                jointParameters HingeJointParameters {
                  position -1189.7948992179845
                  axis 0 1 0
                  anchor 0.05 0 0.02
                }
                endPoint Solid {
                  translation 0.05 0 0.02
                  rotation 0 1 0 4.010309146136937
                  children [
                    USE r6
                  ]
                  name "sr6"
                  boundingObject USE r6
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr7 HingeJoint {
                jointParameters HingeJointParameters {
                  position 1916.6494303777845
                  axis -0.707107 0.707107 0
                  anchor -0.03535 -0.03535 0.02
                }
                endPoint Solid {
                  translation -0.03535 -0.03535 0.02
                  rotation 0.7071067811865475 -0.7071067811865475 0 6.005273619168979
                  children [
                    USE r7
                  ]
                  name "sr7"
                  boundingObject USE r7
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr8 HingeJoint {
                jointParameters HingeJointParameters {
                  position -4974.734965659909
                  axis -0.707107 0.707107 0
                  anchor 0.03535 0.03535 0.02
                }
                endPoint Solid {
                  translation 0.03535 0.03535 0.02
                  rotation -0.7071067811865475 0.7071067811865475 0 1.5477976263234958
                  children [
                    USE r8
                  ]
                  name "sr8"
                  boundingObject USE r8
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              USE WHEEL_TRANS
            ]
            boundingObject Transform {
              translation 0 0 0.02
              rotation 1 0 0 1.5708
              children [
                DEF wheel1 Shape {
                  appearance PBRAppearance {
                    baseColor 0.8 0.8 0.8
                    transparency 0.12999999523162842
                    roughness 1
                    metalness 0
                    emissiveColor 0 1 0
                  }
                  geometry Cylinder {
                    height 0.02
                    radius 0.05
                  }
                }
              ]
            }
            physics Physics {
            }
          }
        }
      ]
      name "wheel 2"
      boundingObject USE DUMMY
      physics Physics {
      }
    }
    DEF SOLID3 Solid {
      rotation 0 1 0 4.18
      children [
        DEF WHEEL3 HingeJoint {
          jointParameters HingeJointParameters {
            position -31.418926898289325
            axis 0 0 1
            anchor 0 0 0.1
          }
          device [
            RotationalMotor {
              name "motor_3"
              maxVelocity 30.3688
              maxTorque 0.847386
            }
            PositionSensor {
              name "position_3"
              resolution 0.00384841
            }
          ]
          endPoint Solid {
            translation 0 0 0.1
            rotation 0 0 -1 6.282276229586446
            children [
              DEF sr1 HingeJoint {
                jointParameters HingeJointParameters {
                  position -666.3805568700485
                  anchor 0 0.05 0.02
                }
                endPoint Solid {
                  translation 0 0.05 0.02
                  rotation -0.9999999999999999 0 0 0.3629143090123414
                  children [
                    USE r1
                  ]
                  name "sr1"
                  boundingObject USE r1
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr2 HingeJoint {
                jointParameters HingeJointParameters {
                  position -1858.0869428270164
                  axis 0.707107 0.707107 0
                  anchor -0.03535 0.03535 0.02
                }
                endPoint Solid {
                  translation -0.03535 0.03535 0.02
                  rotation 0.7071067811865475 0.7071067811865475 0 1.7359080981411965
                  children [
                    USE r2
                  ]
                  name "sr2"
                  boundingObject USE r2
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr3 HingeJoint {
                jointParameters HingeJointParameters {
                  position 197.1667872546652
                  anchor 0 -0.05 0.02
                }
                endPoint Solid {
                  translation 0 -0.05 0.02
                  rotation -0.9999999999999999 0 0 3.8951425750815725
                  children [
                    USE r3
                  ]
                  name "sr3"
                  boundingObject USE r3
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr4 HingeJoint {
                jointParameters HingeJointParameters {
                  position -5.484561604708116
                  axis 0.707107 0.707107 0
                  anchor 0.03535 -0.03535 0.02
                }
                endPoint Solid {
                  translation 0.03535 -0.03535 0.02
                  rotation -0.7071067811865475 -0.7071067811865475 0 5.484561604708116
                  children [
                    USE r4
                  ]
                  name "sr4"
                  boundingObject USE r4
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr5 HingeJoint {
                jointParameters HingeJointParameters {
                  position -2257.320236144439
                  axis 0 1 0
                  anchor -0.05 0 0.02
                }
                endPoint Solid {
                  translation -0.05 0 0.02
                  rotation 0 1 0 4.6264744402119815
                  children [
                    USE r5
                  ]
                  name "sr5"
                  boundingObject USE r5
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr6 HingeJoint {
                jointParameters HingeJointParameters {
                  position -2040.316613322516
                  axis 0 1 0
                  anchor 0.05 0 0.02
                }
                endPoint Solid {
                  translation 0.05 0 0.02
                  rotation 0 -1 0 4.5645737963300865
                  children [
                    USE r6
                  ]
                  name "sr6"
                  boundingObject USE r6
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr7 HingeJoint {
                jointParameters HingeJointParameters {
                  position -3335.9524829857887
                  axis -0.707107 0.707107 0
                  anchor -0.03535 -0.03535 0.02
                }
                endPoint Solid {
                  translation -0.03535 -0.03535 0.02
                  rotation 0.7071067811865476 -0.7071067811865476 0 5.864270180607886
                  children [
                    USE r7
                  ]
                  name "sr7"
                  boundingObject USE r7
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr8 HingeJoint {
                jointParameters HingeJointParameters {
                  position -602.6360489601639
                  axis -0.707107 0.707107 0
                  anchor 0.03535 0.03535 0.02
                }
                endPoint Solid {
                  translation 0.03535 0.03535 0.02
                  rotation -0.7071067811865476 0.7071067811865476 0 0.5497405290764394
                  children [
                    USE r8
                  ]
                  name "sr8"
                  boundingObject USE r8
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              USE WHEEL_TRANS
            ]
            boundingObject Transform {
              translation 0 0 0.02
              rotation 1 0 0 1.5708
              children [
                DEF wheel1 Shape {
                  appearance PBRAppearance {
                    baseColor 0.8 0.8 0.8
                    transparency 0.12999999523162842
                    roughness 1
                    metalness 0
                    emissiveColor 0 1 0
                  }
                  geometry Cylinder {
                    height 0.02
                    radius 0.05
                  }
                }
              ]
            }
            physics Physics {
            }
          }
        }
      ]
      name "wheel 3"
      boundingObject USE DUMMY
      physics Physics {
      }
    }
    DEF BODY Transform {
      translation 0 0.03 0
      rotation 0 0 1 4.712390306100425
      scale 1 0.7 0.7
      children [
        Shape {
          appearance Appearance {
            material Material {
              ambientIntensity 1
              diffuseColor 0.7568627450980392 0.49019607843137253 0.06666666666666667
              shininess 0.4
              specularColor 0.396078 0.409804 0.466667
            }
          }
          geometry IndexedFaceSet {
            coord Coordinate {
              point [
                0 -0.159225 0.085098
                0 -0.153309 -0.095344
                0 -0.143127 0.076928
                0 -0.138185 -0.085488
                0 -0.005915 0.180442
                0 -0.004942 0.162416
                0 0.004942 -0.162416
                0 0.005915 -0.180442
                0 0.138185 0.085488
                0 0.143127 -0.076928
                0 0.153309 0.095344
                0 0.159225 -0.085098
                0.03 -0.143127 0.076928
                0.03 -0.138185 -0.085488
                0.03 -0.004942 0.162416
                0.03 0.004942 -0.162416
                0.03 0.138185 0.085488
                0.03 0.143127 -0.076928
                0.05 -0.159225 0.085098
                0.05 -0.153309 -0.095344
                0.05 -0.005915 0.180442
                0.05 0.005915 -0.180442
                0.05 0.153309 0.095344
                0.05 0.159225 -0.085098
              ]
            }
            coordIndex [
              20, 22, 4, -1, 4, 22, 10, -1, 18, 20, 0, -1, 0, 20, 4, -1, 19, 18, 1, -1, 1, 18, 0, -1, 21, 19, 7, -1, 7, 19, 1, -1, 23, 21, 11, -1, 11, 21, 7, -1, 22, 23, 10, -1, 10, 23, 11, -1, 23, 22, 21, -1, 21, 22, 20, -1, 21, 20, 19, -1, 19, 20, 18, -1, 6, 7, 3, -1, 3, 7, 1, -1, 3, 1, 2, -1, 2, 1, 0, -1, 2, 0, 5, -1, 5, 0, 4, -1, 5, 4, 8, -1, 8, 4, 10, -1, 8, 10, 9, -1, 9, 10, 11, -1, 9, 11, 6, -1, 6, 11, 7, -1, 17, 16, 9, -1, 9, 16, 8, -1, 16, 14, 8, -1, 8, 14, 5, -1, 14, 12, 5, -1, 5, 12, 2, -1, 12, 13, 2, -1, 2, 13, 3, -1, 13, 15, 3, -1, 3, 15, 6, -1, 15, 17, 6, -1, 6, 17, 9, -1, 15, 13, 17, -1, 17, 13, 12, -1, 17, 12, 16, -1, 16, 12, 14, -1
            ]
          }
        }
      ]
    }
    Solid {
      translation 0.1 0.036 0.060000000000000005
      rotation 0 1 0 -0.6544996938995746
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.9372549019607843 0.1607843137254902 0.1607843137254902
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.1
          }
        }
      ]
    }
    DistanceSensor {
      translation 0.13 0.04 0.019999999999999997
      rotation 0 1 0 -0.43633269389957474
      name "distance_sensor2"
      lookupTable [
        0 0 0
        0.2 65535 0
      ]
      resolution 1
    }
    DistanceSensor {
      translation 0.06999999999999999 0.04 0.09999999999999999
      rotation 0 1 0 -0.5672326938995748
      name "distance_sensor1"
      lookupTable [
        0 0 0
        0.2 65535 0
      ]
      aperture 1.57
      resolution 1
    }
  ]
  boundingObject USE BODY
  physics Physics {
  }
  controller "U3_Project_Victor_Xix"
}
